exercises/practice/complex-numbers/complex_numbers_test.go

```diff
@@ -0,0 +1,127 @@
+package complex
+
+import (
+	"math"
+	"testing"
+)
+
+func TestNumber_Real(t *testing.T) {
+	for _, tt := range realTestCases {
+		t.Run(tt.description, func(t *testing.T) {
+			n := Number{tt.in.a, tt.in.b}
+			if got := n.Real(); got != tt.want {
+				t.Errorf("Number({%v, %v}).Real() = %v, want %v", tt.in.a, tt.in.b, got, tt.want)
+			}
+		})
+	}
+}
+
+func TestNumber_Imaginary(t *testing.T) {
+	for _, tt := range imaginaryTestCases {
+		t.Run(tt.description, func(t *testing.T) {
+			n := Number{tt.in.a, tt.in.b}
+			if got := n.Imaginary(); got != tt.want {
+				t.Errorf("Number({%v, %v}).Imaginary() = %v, want %v", tt.in.a, tt.in.b, got, tt.want)
+			}
+		})
+	}
+}
+
+func TestNumber_Add(t *testing.T) {
+	for _, tt := range addTestCases {
+		t.Run(tt.description, func(t *testing.T) {
+			n1 := Number{tt.n1.a, tt.n1.b}
+			n2 := Number{tt.n2.a, tt.n2.b}
+			if got := n1.Add(n2); got.Real() != tt.want.a || got.Imaginary() != tt.want.b {
+				t.Errorf("Number({%v, %v}).Add({%v, %v})\n got: a=%v, b=%v\nwant: a=%v, b=%v", tt.n1.a, tt.n1.b, tt.n2.a, tt.n2.b, got.Real(), got.Imaginary(), tt.want.a, tt.want.b)
```

Comment:
```suggestion
				t.Errorf("Number%+v.Add(%+v)\n got: %+v\nwant: %+v", tt.n1, tt.n2, got, tt.want)
```

Other than the suggestion of starting with `Number%+v` that was also previously suggested, here I also used the `%+v` formatting flag to simplify the code a bit, let me know what you think.